lib/Analysis/Utility.cpp

```diff
@@ -228,6 +228,7 @@ bool isMfmaToDotShortcut(RankedTensorType &srcTy, RankedTensorType &dstTy) {
   // layout when opIdx == 1.
   return mfmaLayout.getWarpsPerCTA()[1] == 1 &&
          dotOperandLayout.getOpIdx() == 0 &&
+         dotOperandLayout.getKWidth() == 8 &&
```

Comment:
Why we need this?

Reply:
This condition makes sure that mfma dot operand layout is compatible with mfma dot output.
This is how threads are distributed across mfma dot output:
![Screenshot from 2023-08-31 16-56-23](https://github.com/ROCmSoftwarePlatform/triton/assets/3631351/f12b5511-9c42-4c33-9c77-99e30f7d36f4)

Matrix elements are grouped in clusters of 32 columns x 8 rows.

Elements of B operand also distributed like this, but number of "rows" (aka k dimension of B operand) could be different depending on instruction used.
For example fp32 will have similar structure, but size of group will be 32x2, i.e. `kWidth` == 2

Reply:
yes, plus 32x32x8 i8 x i8 -> i32 variant